tests/unit/test_client.py

```diff
@@ -673,6 +673,9 @@ def test_no_executor_used_when_no_transfers(self, mock_sdk_blob_client):
     @pytest.mark.parametrize(
         "blob_size, download_offset, download_length, expected_ranges, known_blob_size",
         [
+            # Download an empty blob
+            (0, None, None, ["0--1"], True),
+            (0, None, None, [f"0-{DEFAULT_PARTITION_SIZE - 1}"], False),
```

Comment:
It would also be good to add a cases where the invalid range exception is thrown where:
1. We still raise the exception if the size is larger than 0 bytes
2. We return empty bytes if the size is 0 bytes

I think we should be able to add a new test method similar to how `test_maps_download_exceptions` is set up.